### PR TITLE
[REFACTORING] ResidueDB

### PR DESCRIPTION
**Pull request by timosachsenberg**
# Description

- clear separation between unmodified and modified residues
- removed unneeded critical section
- trimmed down code a bit

# Checklist:
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

### CODE REVIEW

**Comment by timosachsenberg** [comment]
rebuild jenkins

**Comment by timosachsenberg** [comment]
rebuild jenkins

**Comment by timosachsenberg** [comment]
rebuild jenkins

File: src/openms/include/OpenMS/CHEMISTRY/ResidueDB.h [hunk 1]
```diff
@@ -145,47 +146,46 @@ namespace OpenMS
...
-    ResidueDB& operator=(const ResidueDB& aa);
-   
+    ResidueDB& operator=(const ResidueDB& aa) = delete;
+    //@}
+
    // construct all residues 
     void buildResidues_();
     
-    // add residues to the lookup table and add the residue sets to the residues
+    /// creates and adds residues to a lookup tables including the residue set
```
Comment (jpfeuffer) [inline src/openms/include/OpenMS/CHEMISTRY/ResidueDB.h:158]:
```suggestion
    /// creates and adds residues to a lookup table including the residue set
```

File: src/openms/include/OpenMS/CHEMISTRY/ResidueDB.h [hunk 2]
```diff
@@ -145,47 +146,46 @@ namespace OpenMS
...
     void insertResidueAndAssociateWithResidueSet_(Residue* residue, const std::vector<String>& residue_sets);
 
-    /// deletes all sub-instances of the stored data like modifications and residues
+    /// deletes stored data like modifications and residues
     void clear_();
```
Comment (jpfeuffer) [inline src/openms/include/OpenMS/CHEMISTRY/ResidueDB.h:162]: Where is this function defined?

File: src/topp/ProteinQuantifier.cpp [hunk 3]
```diff
@@ -563,7 +563,7 @@ class TOPPProteinQuantifier :
       // if ratiosSILAC-flag is set, print log2-SILACratios. Only if three maps are provided (triple SILAC).
       if (print_SILACratios && ed.getNumberOfSamples() == 3)
       {
-        ConsensusMap::ColumnHeaders::iterator file_it = columns_headers_.begin();
+        //ConsensusMap::ColumnHeaders::iterator file_it = columns_headers_.begin();
```
Comment (jpfeuffer) [inline src/topp/ProteinQuantifier.cpp:566]: Completely remove?

**Review by jpfeuffer — APPROVED** [review]
Looks good. Make sure that it still is 100% thread-safe (running multithreaded tests a lot of times). We had some hard to find errors in this class already.

File: src/openms/include/OpenMS/CHEMISTRY/ResidueDB.h (hunk 2 above)
Comment (timosachsenberg) [reply src/openms/include/OpenMS/CHEMISTRY/ResidueDB.h]: removed

**Comment by timosachsenberg** [comment]
Since c11 the initialization of the static is guaranteed thread-safe. That's why the single critical section can be removed

**Pull request merged**